lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -104,7 +104,11 @@ public function quote($input, $type = ParameterType::STRING)
      */
     public function lastInsertId($name = null)
     {
-        return parent::lastInsertId($name);
+        try {
+            return parent::lastInsertId($name);
+        } catch (\PDOException $exception) {
```

Comment:
Is this covered by a test somewhere?

Reply:
Yes, in https://github.com/doctrine/dbal/pull/3339#discussion_r238083147.